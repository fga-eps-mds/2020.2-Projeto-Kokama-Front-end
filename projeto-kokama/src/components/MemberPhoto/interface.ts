export interface MemberPhotoProps {
  name: string;
  photo: any;
}
