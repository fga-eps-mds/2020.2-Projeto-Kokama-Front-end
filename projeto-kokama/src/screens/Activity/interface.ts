export interface Exercise {
    id: number;
    phrase_portuguese: string;
    phrase_kokama: string;
    options: Array<string>;
}