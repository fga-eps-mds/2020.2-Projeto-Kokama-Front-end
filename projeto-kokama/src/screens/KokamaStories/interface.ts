export interface kokamaStories {
    title: string;
    text: string;
}