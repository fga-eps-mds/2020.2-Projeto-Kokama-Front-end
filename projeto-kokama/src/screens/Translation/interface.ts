export interface Phrase {
    phrase_portuguese: string;
    phrase_kokama: string;
}

export default interface Dictionary {
    word_kokama: string;
    pronunciation_type: string;
    phrases: Phrase[];
    translations: string[];
}