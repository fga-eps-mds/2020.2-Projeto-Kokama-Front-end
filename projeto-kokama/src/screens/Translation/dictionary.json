[
  {
    "word_kokama": "pacha",
    "pronunciation_type": "geral",
    "phrases": [
      {
        "phrase_kokama": "Emete ipiranu <pachakatika> iruataka chiranu.",
        "phrase_portuguese": "Há <centenas> de peixes de toda classe de nomes."
      }
    ],
    "translations": ["cem"]
  },
  {
    "word_kokama": "pachata",
    "pronunciation_type": "geral",
    "phrases": [
      {
        "phrase_kokama": "Nós <esmagamos> a massa para fazer um assento de jarro.",
        "phrase_portuguese": "Nós <esmagamos> a massa para fazer um assento de jarro."
      }
    ],
    "translations": ["aderir", "fixar", "colar"]
  },
  {
    "word_kokama": "pachuri",
    "pronunciation_type": "geral",
    "phrases": [
      {
        "phrase_kokama": "<Pachuri> eyu watsatsa iya.",
        "phrase_portuguese": "O <pía pía> come semente de abiu."
      }
    ],
    "translations": ["pía pía"]
  },
  {
    "word_kokama": "pai",
    "pronunciation_type": "geral",
    "phrases": [
      {
        "phrase_kokama": "Tsa <pai> papa imua",
        "phrase_portuguese": "Meu <tio> é irmão de papai"
      }
    ],
    "translations": ["tio"]
  },
  {
    "word_kokama": "pain",
    "pronunciation_type": "geral",
    "phrases": [
      {
        "phrase_kokama": "Yawiri <pain>pura tsa papa erura.",
        "phrase_portuguese": "<Só> a mandioca traz meu pai."
      }
    ],
    "translations": ["apenas", "puro", "não misturado"]
  },
  {
    "word_kokama": "pararaka",
    "pronunciation_type": "geral",
    "phrases": [
      {
        "phrase_kokama": "<Pararaka> yute kuarachika ya ikanatsen.",
        "phrase_portuguese": "<Espalhe> essa juta em direção ao sol (não na sombra) para secar."
      },
      {
        "phrase_kokama": "Ipira na ikanata <pararaka>.",
        "phrase_portuguese": "Seque o peixe <espalhando-o>. "
      },
      {
        "phrase_kokama": "Tsetsa na tɨata unipu <pararaka>.",
        "phrase_portuguese": "Umedeça as flores <regando-as>."
      }
    ],
    "translations": ["dispersar", "regar", "distribuir"]
  },
  {
    "word_kokama": "penu",
    "pronunciation_type": "feminino",
    "phrases": [
      {
        "phrase_kokama": "<Penu> kumitsa",
        "phrase_portuguese": "<Nós> falamos."
      }
    ],
    "translations": ["nós"]
  },
  {
    "word_kokama": "ini",
    "pronunciation_type": "geral",
    "phrases": [
      {
        "phrase_kokama": "<Ini> kumitsa",
        "phrase_portuguese": "<Nós> falamos."
      }
    ],
    "translations": ["nós"]
  }
]
