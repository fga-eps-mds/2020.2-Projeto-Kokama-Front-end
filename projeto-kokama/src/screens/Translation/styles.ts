import { StyleSheet, Dimensions, Platform, StatusBar } from "react-native";

// Constants and Global Variables
const window = Dimensions.get("window");
const screen = Dimensions.get("screen");
const STATUSBAR_HEIGHT = Platform.OS === "ios" ? 20 : StatusBar.currentHeight;

const styles = StyleSheet.create({
  container: {
    maxWidth: screen.width,
    height: window.height,
    backgroundColor: "#F0F0F0",
  },

  logoArea: {
    flexDirection: "column",
    marginTop: STATUSBAR_HEIGHT + 5,
    borderBottomWidth: 1,
    borderColor: "#BBB",
    width: "100%",
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "#F0F0F0",
  },
  logo: {
    backgroundColor: "#fff",
    borderRadius: 100,
    height: 50,
    width: 50,
  },
  windowName: {
    fontSize: 20,
    marginBottom: 5,
  },
  changeLanguage: {
    height: 55,
    flexDirection: "row",
    borderBottomWidth: 1,
    borderColor: "#BBB",
    backgroundColor: "#F0F0F0",
    justifyContent: "space-between",
    alignItems: "center",
  },
  originLanguageArea: {
    flex: 3,
  },
  originLanguage: {
    textAlign: "left",
    marginLeft: 24,
    fontSize: 22,
  },
  languageExchangeArea: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  destLanguageArea: {
    flex: 3,
  },
  destLanguage: {
    textAlign: "right",
    marginRight: 24,
    fontSize: 22,
  },
  userInput: {
    width: "100%",
    height: 60,
    flexDirection: "row",
    borderTopWidth: 1,
    borderBottomWidth: 1,
    borderColor: "#BBB",
    backgroundColor: "#FFF",
    alignItems: "center",
    justifyContent: "space-between",
  },
  textBox: {
    flex: 6,
    textAlign: "left",
    fontSize: 20,
    paddingLeft: 23,
  },
  symbolArea: {
    flex: 1,
    height: "65%",
    marginRight: 15,
    borderWidth: 2.5,
    borderColor: "#ff6363",
    borderRadius: 15,
    alignItems: "center",
    justifyContent: "center",
  },
  symbol: {
    fontSize: 25,
  },
  translationArea: {
    width: "100%",
    backgroundColor: "#FFF",
    borderBottomWidth: 1.5,
    borderTopWidth: 1,
    marginTop: 5,
    paddingBottom: 10,
    borderColor: "#BBB",
    alignItems: "center",
  },
  translatedWord: {
    height: 80,
    width: "80%",
    alignSelf: "center",
    textAlignVertical: "center",
    textAlign: "left",
    fontSize: 28,
    paddingLeft: 23,
    marginBottom: 20,
    borderBottomWidth: 1.5,
    borderColor: "#BBB",
  },
  exampleArea: {
    width: "90%",
    marginVertical: 10,
  },
  label: {
    height: 30,
    fontWeight: "bold",
    fontSize: 17,
  },
  examples: {
    width: "100%",
    marginHorizontal: 10,
    justifyContent: "flex-start",
    alignItems: "flex-start",
    flexDirection: "row",
  },
  examplesText: {
    fontSize: 20,
    color: "#000",
    height: "100%",
    flexWrap: "wrap",
  },
  examplesWord: {
    fontSize: 20,
    color: "#FF6363",
    height: "100%",
    flexWrap: "wrap",
  },
  historyArea: {
    flexDirection: "row",
    height: 40,
    borderBottomWidth: 1.5,
    borderTopWidth: 1.5,
    marginTop: 5,
    borderColor: "#BBB",
    backgroundColor: "#FFF",
  },
  historyText: {
    width: "100%",
    textAlignVertical: "center",
    textAlign: "left",
    fontSize: 20,
    paddingLeft: 23,
    color: "#a5a5a5",
  },
  historyWordsArea: {
    flexDirection: "column",
    backgroundColor: "#FFF",
    borderBottomWidth: 1.5,
    borderColor: "#BBB",
  },
  historyWords: {
    width: "100%",
    textAlignVertical: "center",
    textAlign: "left",
    fontSize: 18,
    paddingLeft: 15,
    color: "#555",
  },
});

export default styles;