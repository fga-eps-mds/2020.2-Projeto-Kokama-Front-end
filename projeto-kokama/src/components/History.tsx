import React from "react";
import { StyleSheet, View, Text, TouchableWithoutFeedback, ScrollView } from "react-native";
import Colors from "../assets/Colors";
import { HistoryTuple } from "../screens/Translation/interface";
import HistoryTitle from "./HistoryTitle";
import { capitalizeFirstLetter } from "../utils/translation";

const styles = StyleSheet.create({
  Area: {
    flexDirection: "column",
    backgroundColor: Colors.WHITE,
    borderBottomWidth: 1.5,
    borderColor: Colors.DARK_GRAY,
  },
  WordsArea: {
    width: "100%",
    textAlignVertical: "center",
    textAlign: "left",
    fontSize: 18,
    borderBottomWidth: 0.5,
    borderColor: Colors.DARK_GRAY,
    paddingLeft: 23,
    paddingVertical: 7,
  },
  Words: {
    width: "100%",
    textAlignVertical: "center",
    textAlign: "left",
    fontSize: 18,
    color: Colors.HISTORY_WORD_TEXT,
  },
});

interface Props {
  isEnabled: boolean;
  data: HistoryTuple[];
  onPressTitle: () => void;
  translateFrom: string;
  onPressWord: (kokamaWord:string, originLanguage:string) => void;
}

const History = (props: Props) => {
  return (
    <View>
      <HistoryTitle
        isEnabled={props.isEnabled}
        data={props.data}
        onPressTitle={props.onPressTitle}
      />
      
      {props.data.length > 0 && props.isEnabled && (
        <ScrollView style={styles.Area}>
          {props.data.map((word: HistoryTuple, index: number) => (
            <TouchableWithoutFeedback
              key={index}
              onPress={() => props.onPressWord(word.kokama, props.translateFrom)}
            >
              <View style={styles.WordsArea}>
                <Text style={[styles.Words, { fontWeight: "bold", fontSize: 20 }]}>
                  {capitalizeFirstLetter(word.kokama)}
                </Text>
                <Text style={styles.Words}>
                  {capitalizeFirstLetter(word.portuguese)}
                </Text>
              </View>
            </TouchableWithoutFeedback>
          ))}
        </ScrollView>
      )}
    </View>
  );
};

export default History;
