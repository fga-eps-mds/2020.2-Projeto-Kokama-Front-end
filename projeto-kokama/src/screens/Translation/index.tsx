import {
  View,
  Text,
  TextInput,
  TouchableWithoutFeedback,
  ScrollView,
  SafeAreaView,
} from "react-native";
import React, { useState } from "react";
import SyncStorage from "sync-storage";
import Icon from "react-native-vector-icons/AntDesign";
import HighlightText from "@sanar/react-native-highlight-text";
import { Dictionary, Phrase, HistoryTuple } from "./interface";
import {
  PORTUGUESE,
  KOKAMA,
  HISTORY_SIZE,
} from "../../config/constants";
import translationStyle from "./styles";
import {
  capitalizeFirstLetter,
  removeStringFromMarkers,
} from "../../utils/translation";
import Colors from "../../assets/Colors";
import TopMenu from "../../components/TopMenu";
import History from "../../components/History";
import styles from "./styles";

let historyArray: Array<HistoryTuple> = SyncStorage.get("history") || [];

const Translation = () => {
  const [translation, setTranslation] = useState("");
  const [originLanguage, setOriginLanguage] = useState(PORTUGUESE);
  const [destLanguage, setDestLanguage] = useState(KOKAMA);
  const [historyIsEnabled, setHistoryIsEnabled] = useState(false);
  const toggleHistory = () => {
    if (historyArray.length > 0) {
      setHistoryIsEnabled((previousState) => !previousState);
    }
  }

  function exchangeLanguage() {
    let temp = originLanguage;
    setOriginLanguage(destLanguage);
    setDestLanguage(temp);
  }

  function insertSymbol() {
    setTranslation(translation + "ɨ");
  }

  // When a word with translation is found, store it in history
  // An Array Object is saved in storage with a "history" key
  function addHistoryWord(
    kokamaWord: string,
    portugueseWords: Array<string>
  ) {
    // Format portuguese words string for presentation
    let concatPortuguese: string = portugueseWords.join(", ");

    // Create new history element to add to the history array and storage
    let historyWord: HistoryTuple = {
      kokama: kokamaWord,
      portuguese: concatPortuguese,
    };
    // If the element to be added is already in history, delete the old one
    historyArray.forEach((word, index) => {
      if (word.kokama == historyWord.kokama) {
        historyArray.splice(index, 1);
      }
    });
    // Add the new history element at the beginning of the history array
    historyArray.unshift(historyWord);
    // Remove the oldest element if history reach max size
    if (historyArray.length > HISTORY_SIZE) {
      historyArray.pop();
    }
    // Add the updated history array as an object to storage
    SyncStorage.set("history", historyArray);
  }

  // Set the text input equal to the kokama word the user pressed inside history
  function translateHistoryWord(word: string, language: string) {
    setTranslation(capitalizeFirstLetter(word));
    // Automatically changes to kokama
    if (language !== KOKAMA) {
      exchangeLanguage();
    }
  }

  // Check and return from dictionary word element that match the kokama user input
  function getKokamaElement(userInput: string) {
    let kokamaElement: Array<Dictionary> = [];
    for (let element of SyncStorage.get("dictionary")) {
      if (userInput.toLowerCase() == element.word_kokama.toLowerCase()) {
        // Add matching word to the return array
        kokamaElement.push(element);
        // Add matching word to history
        addHistoryWord(
          element.word_kokama,
          element.translations
        );
        break;
      }
    }

    return kokamaElement;
  }

  // Check and return from dictionary word elements that match the portuguese user input
  function getPortugueseElement(userInput: string) {
    let portugueseElements: Array<Dictionary> = [];

    SyncStorage.get("dictionary").forEach((element: Dictionary) => {
      for (let word of element.translations) {
        if (userInput.toLowerCase() == word.toLowerCase()) {
          // Add matching word to the return array
          portugueseElements.push(element);
          // Add matching word to history
          addHistoryWord(
            element.word_kokama,
            element.translations
          );
        }
      }
    });

    return portugueseElements;
  }

  // Returns all dictionary elements that matches the user input, based on the language searched
  function getDictionaryElements(language: string, userInput: string) {
    let dictionaryElements: Array<Dictionary> = [];

    if (language == KOKAMA) {
      let kokamaElement = getKokamaElement(userInput);
      dictionaryElements = dictionaryElements.concat(kokamaElement);
    } else {
      let portugueseElements = getPortugueseElement(userInput);
      dictionaryElements = dictionaryElements.concat(portugueseElements);
    }

    return dictionaryElements;
  }

  // For a given dictionary element(word), return its kokama and portuguese words for presentation
  function getWords(language: string, word: Dictionary) {
    if (language == KOKAMA) {
      return word.translations;
    } else {
      return [word.word_kokama];
    }
  }

  // For a given dictionary element array, return its respective translation for presentation
  function getTranslations(
    language: string,
    dictionaryElements: Array<Dictionary>
  ) {
    let translatedWords: string = "";

    for (let element of dictionaryElements) {
      let words: Array<string> = getWords(language, element);
      translatedWords = words.join(", ");
    }

    return translatedWords;
  }

  // For a given dictionary element array, return its respective example phrases for presentation
  function getPhrases(dictionaryElements: Array<Dictionary>) {
    let phrases: Array<Phrase> = [];
    for (let element of dictionaryElements) {
      phrases = phrases.concat(element.phrases);
    }

    return phrases;
  }

  // Main translation function
  // Return TSX visualization of the translation
  function Translate(language: string, userInput: string) {
    let translationList: Array<Dictionary> = getDictionaryElements(
      language,
      userInput
    );

    let words = getTranslations(language, translationList);
    let phrases: Array<Phrase> = getPhrases(translationList);

    return (
      <View>
        {/* Warning if the user input does not match any word in dictionary */}
        {phrases.length === 0 && translation !== "" && (
          <View style={{ alignItems: "center" }}>
            <Text style={{ textAlign: "center" }}>Tradução não encontrada</Text>
          </View>
        )}
        {/* If a translation is found, present the translations and phrases */}
        {phrases.length > 0 && (
          <View style={translationStyle.translationArea}>
            {/* Presentation of the translations words */}
            <Text style={translationStyle.translatedWord}>
              {capitalizeFirstLetter(words)}
            </Text>
            {phrases.map((phrase, index) => (
              <View style={translationStyle.exampleArea} key={index}>
                {/* Phrase kokama */}
                <HighlightText
                  style={translationStyle.examplesText}
                  highlightStyle={{ color: Colors.RED }}
                  searchWords={[
                    removeStringFromMarkers(phrase.phrase_kokama, "<", ">"),
                  ]}
                  textToHighlight={phrase.phrase_kokama
                    .replace("<", "")
                    .replace(">", "")}
                />
                {/* Phrase portuguese */}
                <HighlightText
                  style={translationStyle.examplesText}
                  highlightStyle={{ color: Colors.RED }}
                  searchWords={[
                    removeStringFromMarkers(phrase.phrase_portuguese, "<", ">"),
                  ]}
                  textToHighlight={phrase.phrase_portuguese
                    .replace("<", "")
                    .replace(">", "")}
                />
              </View>
            ))}
          </View>
        )}
      </View>
    );
  }

  return (
    <SafeAreaView>
      <ScrollView
        style={translationStyle.container}
        keyboardShouldPersistTaps={"always"}
      >

        {/* Change language area */}
        <View style={translationStyle.changeLanguage}>
          {/* First language */}
          <View style={[translationStyle.originLanguageArea]}>
            <Text style={translationStyle.originLanguage}>
              {originLanguage}
            </Text>
          </View>

          {/* Change language icon */}
          <View style={translationStyle.languageExchangeArea}>
            <TouchableWithoutFeedback onPress={exchangeLanguage}>
              <Icon name="swap" size={40} />
            </TouchableWithoutFeedback>
          </View>

          {/* Second Language */}
          <View style={[translationStyle.destLanguageArea]}>
            <Text style={translationStyle.destLanguage}>{destLanguage}</Text>
          </View>
        </View>

        {/* Text box for the user entry */}
        <View style={translationStyle.userInput}>
          <TextInput
            style={translationStyle.textBox}
            placeholder="Toque para digitar"
            onChangeText={(input) => setTranslation(input)}
            defaultValue={translation}
          />

          <TouchableWithoutFeedback onPress={insertSymbol}>
            <View style={translationStyle.symbolArea}>
              <Text style={translationStyle.symbol}>ɨ</Text>
            </View>
          </TouchableWithoutFeedback>
        </View>

        {/* Translate answer */}
        {SyncStorage.get("dictionary") !== undefined && (
          <View>{Translate(originLanguage, translation)}</View>
        )}

        <History
          isEnabled={historyIsEnabled}
          data={historyArray}
          onPressTitle={toggleHistory}
          onPressWord={translateHistoryWord}
          translateFrom={originLanguage}
        />
      </ScrollView>
    </SafeAreaView>
  );
};

export default Translation;
