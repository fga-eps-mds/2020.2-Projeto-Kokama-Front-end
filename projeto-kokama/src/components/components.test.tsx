import React from 'react';
import renderer from 'react-test-renderer';
import History from './History';
import { HistoryTuple } from "../screens/Translation/interface";
import {act}  from 'react-test-renderer';

interface Props {
    isEnabled: boolean;
    data: HistoryTuple[];
    onPressTitle: () => void;
    translateFrom: string;
    onPressWord: (kokamaWord:string, originLanguage:string) => void;
}

let props:Props
beforeEach(() => {
    props = {
        isEnabled: true,
        data: [{
            kokama: "pacha",
            portuguese: "cem",
          }],
        onPressTitle: () => {},
        translateFrom: "portugues",
        onPressWord:() =>{},
    };
  });


test('renders correctly', () => {
  const tree = renderer.create(<History {... props}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
  let tree = renderer.create(<History {... props}/>);
  const TouchableWithoutFeedback = tree.root.findByProps({testID:'history'}).props;
  act(() => TouchableWithoutFeedback.onPress());
});
