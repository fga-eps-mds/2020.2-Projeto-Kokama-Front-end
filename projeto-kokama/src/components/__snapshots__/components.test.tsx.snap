// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomWidth": 1.5,
        "borderColor": "#E4DFDF",
        "borderTopWidth": 1.5,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "space-between",
        "marginTop": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#A5A5A5",
          "fontSize": 20,
          "paddingLeft": 23,
          "textAlign": "left",
        }
      }
    >
      Hist√≥rico
    </Text>
    <View
      style={
        Object {
          "marginRight": 25,
        }
      }
    >
      <Text />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderBottomWidth": 1.5,
        "borderColor": "#E4DFDF",
        "flexDirection": "column",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderBottomWidth": 0.5,
          "borderColor": "#E4DFDF",
          "fontSize": 18,
          "paddingLeft": 23,
          "paddingVertical": 7,
          "textAlign": "left",
          "textAlignVertical": "center",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#555555",
              "fontSize": 18,
              "textAlign": "left",
              "textAlignVertical": "center",
              "width": "100%",
            },
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Pacha
      </Text>
      <Text
        style={
          Object {
            "color": "#555555",
            "fontSize": 18,
            "textAlign": "left",
            "textAlignVertical": "center",
            "width": "100%",
          }
        }
      >
        Cem
      </Text>
    </View>
  </View>
</View>
`;
